### PR TITLE
New traces: `ScanExecution` and `RuleScanTime`

### PR DESCRIPTION
**Pull request by michaelcfanning**
Add rule timing trace knob (and emit trace data when enabled).

Add general execution trace flag (with no implementation).

### CODE REVIEW

File: src/ReleaseHistory.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 * BREAKING: Correct casing of `LogMissingreportingConfiguration` helper to `LogMissingReportingConfiguration`. [#2599](https://github.com/microsoft/sarif-sdk/pull/2599)
 * BREAKING: Change type of `MaxFileSizeInKilobytes` from int to long in `IAnalysisContext` and other classes.  [#2599](https://github.com/microsoft/sarif-sdk/pull/2599)
 * BREAKING: For `Guid` properties defined in SARIF spec, updated Json schema to use `uuid`, and updated C# object model to use `Guid?` instead of `string`. [#2555](https://github.com/microsoft/sarif-sdk/pull/2555)
+* BUGFIX	 : Another attempt to resolve 'InvalidOperationException' with message `Collection was modified; enumeration operation may not execute` in `MultithreadedAnalyzeCommandBase`, raised when analyzing with the `--hashes` switch. [#2459](https://github.com/microsoft/sarif-sdk/pull/2549). There was a previous attempt to fix this in [#2447](https://github.com/microsoft/sarif-sdk/pull/2447).
```
Comment (yongyan-gh) [inline src/ReleaseHistory.md:8]: >	 [](http://example.com/codeflow?start=8&length=1)

nit: replace the tab with a whitespace

Comment (yongyan-gh) [inline src/ReleaseHistory.md:8]: this line is duplicated of line 13

File: src/ReleaseHistory.md [hunk 2]
```diff
@@ -15,6 +16,9 @@
 * BUGFIX  : Update `merge` command to properly produce runs by tool and version when passed the `--merge-runs` argument. [#2488](https://github.com/microsoft/sarif-sdk/pull/2488)
 * BUGFIX  : Eliminate `IOException` and `DirectoryNotFoundException` exceptions thrown by `merge` command when splitting by rule (due to invalid file characters in rule ids). [#2513](https://github.com/microsoft/sarif-sdk/pull/2513)
 * BUGFIX  : Fix classes inside NotYetAutoGenerated folder missing `virtual` keyword for public methods and properties, by regenerate and manually sync the changes. [#2537](https://github.com/microsoft/sarif-sdk/pull/2537)
+* FEATURE : Provide mechanism to populate `SarifLogger` with a `FileRegionsCache` instance.
```
Comment (yongyan-gh) [inline src/ReleaseHistory.md:19]: line 19 is a copy of line 30, line 20 is copy of line 31. seems like a merging issue

Comment (michaelcfanning) [reply src/ReleaseHistory.md]: Yes! Thank you for the careful review.

**Pull request merged**